eng/intellisense.targets

```diff
@@ -0,0 +1,115 @@
+<Project>
+  <PropertyGroup>
+    <_UnsupportedTargetFrameworkError></_UnsupportedTargetFrameworkError>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="$(TargetFramework.Contains('-'))">
+    <_OriginalTargetFramework>$(TargetFramework)</_OriginalTargetFramework>
+    <TargetFrameworkSuffix>$(TargetFramework.SubString($([MSBuild]::Add($(TargetFramework.IndexOf('-')), 1))))</TargetFrameworkSuffix>
+    <TargetFramework>$(TargetFramework.SubString(0, $(TargetFramework.IndexOf('-'))))</TargetFramework>
+  </PropertyGroup>
+
+    <!-- Split $(TargetFramework) (e.g. net45) into short identifier and short version (e.g. 'net' and '45'). -->
+  <PropertyGroup Condition="'$(TargetFramework)' != '' and !$(TargetFramework.Contains(',')) and !$(TargetFramework.Contains('+'))">
+   <_ShortFrameworkIdentifier>$(TargetFramework.TrimEnd('.0123456789'))</_ShortFrameworkIdentifier>
+   <_ShortFrameworkVersion>$(TargetFramework.Substring($(_ShortFrameworkIdentifier.Length)))</_ShortFrameworkVersion>
+  </PropertyGroup>
+
+  <!-- Versions with dots are taken as is and just given leading 'v'. -->
+  <PropertyGroup Condition="'$(TargetFrameworkVersion)' == '' and '$(_ShortFrameworkVersion)' != '' and $(_ShortFrameworkVersion.Contains('.'))">
+    <TargetFrameworkVersion>v$(_ShortFrameworkVersion)</TargetFrameworkVersion>
+  </PropertyGroup>
+
+  <!-- Versions with no dots and up to 3 characters get leading 'v' and implicit dots between characters. -->
+  <PropertyGroup Condition="'$(TargetFrameworkVersion)' == '' and '$(_ShortFrameworkVersion)' != ''">
+    <TargetFrameworkVersion Condition="$(_ShortFrameworkVersion.Length) == 1">v$(_ShortFrameworkVersion[0]).0</TargetFrameworkVersion>
```

Comment:
Should we compare this as strings? Like:
`'$(_ShortFrameworkVersion.Length)' == '1'`. I haven't needed to do numerics comparison in MSBuild so I'm not sure.

Reply:
This is just exact copy of the code that sdk uses to parse tfms. We need to re do it here bcoz the original sdk code gets executed before we set our tfm property.

Reply:
Can we set the tfm property in a props/targets file that gets imported before the common targets/props? There is a property that you can set so the SDK imports your targets before theirs. 

There are 2 properties that can help do that: `CustomBeforeMicrosoftCommonTargets` and `AdditionalProjectsBeforeMicrosoftCommonTargets`. https://github.com/microsoft/msbuild/pull/2014

Reply:
I tried to hook into these properties. but they are still late. TargetInference gets imported in "BeforeMicrosoftCommonTargets" and CustomBeforeMicrosoftCommonTargets targets are imported before "MicrosoftCommonTargets"

Reply:
Hmm, that's unfortunate. @ericstj any suggestions for that?

Reply:
We already looked at the hooks and couldn't see anywhere to get in targets that are before the SDK parsing, which is why we're duplicating it.  Here's a sample PP from a clean new project that illustrates this: https://gist.github.com/ericstj/c213e1243e5558d1346d84034920abea#file-out-pp-L654-L900

Reply:
Ok, that's unfortunate 😢

Reply:
Yeah, we'll talk about this with SDK / MSBuild / Nuget to see if we can get a better story in the future.